src/runway/cfngin/hooks/utils.py

```diff
@@ -1,14 +1,186 @@
 """Hook utils."""
 import collections
+import errno
 import logging
 import os
+import re
+import shlex
+import shutil
+import subprocess
 import sys
+import tempfile
+from contextlib import contextmanager
 
-from runway.util import load_object_from_string
+from runway.util import load_object_from_string, change_dir
 
 LOGGER = logging.getLogger(__name__)
 
 
+def filter_commands(commands):
+    """Remove empty commands."""
+    return [cmd for cmd in commands if cmd and len(cmd) > 0]
+
+
+def merge_commands(commands):
+    """Merge command lists."""
+    cmds = filter_commands(commands)
+    if not cmds:
+        raise ValueError('Expected at least one non-empty command')
+    if len(cmds) == 1:
+        return cmds[0]
+
+    script = ' && '.join([shlex.quote(cmd) for cmd in cmds])
+    return ['/bin/sh', '-c', script]
```

Comment:
this can be cut down to

```suggestion
def merge_commands(commands):
    """Merge command lists."""
    cmds = [shlex.quote(cmd) for cmd in commands if cmd]
    if not cmds:
        raise ValueError('Expected at least one non-empty command')
    return ['/bin/sh', '-c', ' && '.join(cmds)]
```